### PR TITLE
Windows rewrite: Part 9 - Completely refactor Main Menu actions, fix reopen, fix connection UI

### PR DESCRIPTION
**Pull request by Kaspik**
## Changes:
- revamp menu actions to SPAppController instead of SPDatabaseDocument as SPDatabaseDocument is not responder anymore
- fix UI on connection screen
- fix reopening miniatuzired window

### CODE REVIEW

File: Interfaces/MainMenu.xib [hunk 1]
```diff
@@ -636,11 +625,6 @@
                                     <action selector="removeDatabase:" target="-1" id="659"/>
                                 </connections>
                             </menuItem>
-                            <menuItem title="Drop Database..." alternate="YES" id="1207">
-                                <connections>
-                                    <action selector="removeDatabase:" target="-1" id="1208"/>
-                                </connections>
-                            </menuItem>
```
Comment (Jason-Morcos) [inline Interfaces/MainMenu.xib:643]: Removed drop database menu item?

File: Source/Controllers/MainViewControllers/ConnectionView/SPConnectionController.m [hunk 2]
```diff
@@ -736,13 +736,13 @@ - (IBAction)updateKeyLocationFileVisibility:(id)sender
...
-	[connectionSplitView setPosition:[[[databaseConnectionView subviews] firstObject] frame].size.width ofDividerAtIndex:0];
-	[connectionSplitView setDelegate:self];
+//	if ([dbDocument getConnection]) {
+//		return;
+//	}
+//
+//	[connectionSplitView setDelegate:nil];
+//	[connectionSplitView setPosition:[[[databaseConnectionView subviews] firstObject] frame].size.width ofDividerAtIndex:0];
+//	[connectionSplitView setDelegate:self];
 }
```
Comment (Jason-Morcos) [inline Source/Controllers/MainViewControllers/ConnectionView/SPConnectionController.m:746]: I believe this was a @jamesstout fix for some earlier bug that ya commented out here (and in the splitViewDidResizeSubviews block below). Why the comment out?

**Review by Jason-Morcos — APPROVED** [review]

File: Interfaces/MainMenu.xib (hunk 1 above)
Comment (Kaspik) [reply Interfaces/MainMenu.xib:643]: There were two `Remove database` and `Drop database`. Both calling the same function.

File: Source/Controllers/MainViewControllers/ConnectionView/SPConnectionController.m (hunk 2 above)
Comment (Kaspik) [reply Source/Controllers/MainViewControllers/ConnectionView/SPConnectionController.m:746]: Nah the fix was with nil coordinates in SPAppController:

```
NSUserDefaults *prefs = [NSUserDefaults standardUserDefaults];

        // fake the dbViewInfoPanelSplit being open
        NSMutableArray *dbViewInfoPanelSplit = [[NSMutableArray alloc] initWithCapacity:2];
        [dbViewInfoPanelSplit addObject:@"0.000000, 0.000000, 359.500000, 577.500000, NO, NO"];
        [dbViewInfoPanelSplit addObject:@"0.000000, 586.500000, 359.500000, 190.500000, NO, NO"];
        [prefs setObject:dbViewInfoPanelSplit forKey:@"NSSplitView Subview Frames DbViewInfoPanelSplit"];
```

Removing this fixes an issue where every new tab has width `2*leftPane.width` of the previousTab

**Pull request merged**